### PR TITLE
Rubber Pig added

### PR DESCRIPTION
**Pull request by komunre**
## About The Pull Request
Adds oinking rubber pig. Costs only 99 inside trade program.

oink

## Why It's Good For The Game
Because I got paid.

## Changelog
:cl:
add: Rubber pig that oinks
/:cl:

### CODE REVIEW

**Comment by ValoTheValo** [comment]
Firstly there is absolutely no reason for this to be made, secondly it will screw up loot and make all these items more common, thirdly, one and possibly more items price changes are direct violation to contributing guidelines, you should try reading them.
https://wiki.cev-eris.com/Content_GuidelinesEn

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -669,3 +669,47 @@
...
+		'sound/items/oink2.ogg',
+		'sound/items/oink3.ogg',
+		'sound/items/oink4.ogg',
+		'sound/items/oink5.ogg',
+		'sound/items/oink6.ogg'
+	)
+
+// Oink code
+/obj/item/toy/rubber_pig/proc/oink()
+	playsound(src, pick(oinks), 50, 1)
```
[the remark this answers is not in the document]
Comment (komunre) [reply code/game/objects/items/toys.dm:693]: fair, will be fixed.

**Comment by komunre** [comment]
> make all these items more common
Does it really matter? We have a whole section of toys in Trade program, that even look like weapons for quite a cheap price, even cheaper than this pig.
If you wish to - sure, we can remove it from trading, but I do not really see a difference.

**Comment by SirRichardFrancis** [comment]
> Does it really matter?

No, it's a meme copypasta.

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -669,3 +669,47 @@
...
 	slot_flags = SLOT_BELT
+
+// That one funny oinking pig.
+/obj/item/toy/rubber_pig
+	name = "rubber pig"
+	desc = "Rubber pig that oinks when squeezed"
+	icon = 'icons/obj/rubber_pig.dmi'
+	icon_state = "icon"
+	item_state = "rubber_pig"
+
```
Comment (SirRichardFrancis) [inline code/game/objects/items/toys.dm:680]:
```suggestion

	var/cooldown = 0.5 SECONDS
	var/last_used = 0
```

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -669,3 +669,47 @@
...
+/obj/item/toy/rubber_pig/throw_impact(atom/impact_atom)
+	src.visible_message(SPAN_NOTICE("Rubber pig oinks, as it impacts with surface."))
+	oink()
+
+// Oinker pick up
+/obj/item/toy/rubber_pig/MouseDrop(over_object, src_location, over_location)
+	..()
+	if (over_object == usr && in_range(src, usr))
+		if (!ishuman(usr)) return
+		usr.put_in_active_hand(src)
```
Comment (SirRichardFrancis) [inline code/game/objects/items/toys.dm:715]:
```suggestion
/obj/item/toy/rubber_pig/MouseDrop(over_object, src_location, over_location)
	..()
	var/mob/living/carbon/human/H = usr
	if(istype(H) && over_object == H && in_range(src, H))
		H.put_in_active_hand(src)

```

**Review by SirRichardFrancis — CHANGES_REQUESTED** [review]

**Comment by ArmolitskiyCarpoeb** [comment]
> Rubber Pigs deal "melee" damage instead of "bullet" damage, this means that for armor, riot armor would be much more effective against Pigs than the bullet-proof vest. This should be noted, since "melee" armor is the lowest value on a good deal of armor vests. A defense against certain pig weaponry is questionable, the damage is superficial at most, tramadol or oxycodrone should negate it. Yet armor with good "melee" values (such as the commander's armored coat) doesn't appear effective against as it should against rubber, depending on the caliber used. A rubber .50AE caliber from the desert oinker, and the Commander's special .45 "Boar" with rubber can output a pain critical in two connected pigs. The Commander's special weapon outputs the highest agony damage of any other weapon I've so far seen with eighty, with another ten brute on top. Command weaponry has such a damage output with rubbers, that you will output 85 damage on simple mobs vs. the 50 damage you'd deal with lethals. That's a dead Fuhrer in three pigs, instead of four. A culminative of 255 damage on the unarmored target, such an attack on a living carbon would absolutely oink their face hard onto the floor. Keeping them in stun-lock would be sustainable, and not endanger the life of the one being put down. The defense against such an assault should be clear, PIG suits with hardsuit chem implanters, as always

unknown-38-1

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -669,3 +669,52 @@
 	item_state = "inflatable"
 	icon = 'icons/inventory/belt/icon.dmi'
 	slot_flags = SLOT_BELT
+
+// That one funny oinking pig.
+/obj/item/toy/rubber_pig
+	name = "rubber pig"
+	desc = "Rubber pig that oinks when squeezed"
```
Comment (Firefox13) [inline code/game/objects/items/toys.dm:676]:
```suggestion
	desc = "Rubber pig that oinks when squeezed."
```

**Review by SirRichardFrancis — APPROVED** [review]

**Review by TheShown911 — APPROVED** [review]

**Review by hyperioo — APPROVED** [review]
I added a picture of the pig to the first message for Rosa's changelog

**Pull request merged**